### PR TITLE
Chore: TIP - Add staff ids

### PR DESCRIPTION
**Pull request by ZarishIqbal**
### This PR contains
- [x] add `tipUniqueId` in staff type
- [x] create hook `useTIPAccess` for TIP Access
- [x] add unique Id field to teacher profile

### CODE REVIEW

File: web/src/pages/staff/create.tsx [hunk 1]
```diff
@@ -97,7 +96,18 @@ export const CreateOrUpdateStaff = () => {
 		if (!isValidPhone(profile.Phone) && profile.Phone !== '') {
 			return toast.error('Please provide correct phone number!')
 		}
+		if (profile.tipUniqueId) {
+			let errorMessage = 'This TIP Id is already assigned to an existing teacher'
+
+			for (const teacher of Object.values(staff ?? {})) {
+				const isDuplicateId =
+					teacher.tipUniqueId === state.profile.tipUniqueId && profile.id !== teacher.id
```
Comment (mudssrali) [inline web/src/pages/staff/create.tsx:104]: why we're anding here?
```
&& profile.id !== teacher.id
```

File: web/src/pages/staff/create.tsx [hunk 2]
```diff
@@ -97,7 +96,18 @@ export const CreateOrUpdateStaff = () => {
...
 
+				if (isDuplicateId) {
+					return toast.error(errorMessage)
+				}
```
Comment (mudssrali) [inline web/src/pages/staff/create.tsx:108]: instead of returning from for..of, would be best if you break the loop and check outside and return

**Review by mudssrali — CHANGES_REQUESTED** [review]
Thanks looks good to me, Thanks @ZarishIqbal however a change and a question in the comments

File: web/src/pages/staff/create.tsx (hunk 1 above)
Comment (ZarishIqbal) [reply web/src/pages/staff/create.tsx:104]: To remove the teacher we are editing at the moment?

**Pull request merged**